src/containers/Console.js

```diff
@@ -35,6 +35,7 @@ function mapDispatchToProps(dispatch) {
   return {
     onClearConsoleEntries() {
       dispatch(clearConsoleEntries());
+      dispatch(focusLine('console', 0, 0));
```

Comment:
This is a totally reasonable instinct, but in the Popcode codebase we tend to think about Redux actions more like _events_ and less like _methods_. Which is to say, an action should convey **what happened**, not instruct **what to do**. Actions that are formulated in the imperative tense typically convey a direct user intent.

Given the above, it’s usually if not always an antipattern to call two actions synchronously in a row. This would imply that two different things happened at the exact same time, which is unlikely.

Instead, we should use a single action to convey what happened—in this case `clearConsoleEntries()` is appropriate, because that is the direct user intent. Then each reducer can decide how the state it’s concerned with should change based on that action. In this case, the `ui` reducer would update the `requestedFocusedLine` property appropriately when consuming the `CLEAR_CONSOLE_ENTRIES` action.

This definitely isn’t the only way to think about the way actions are structured in a Redux application, but having tried out the more “imperative” approach of actions more directly describing the changes to state that they want to effect, I’ve found that the fully reactive approach is easier to reason about and makes the code more nimble as it develops over time.